spec/requests/applicants/omniauth_callbacks_spec.rb

```diff
@@ -0,0 +1,88 @@
+require 'rails_helper'
+
+RSpec.describe 'applicants omniauth call back', type: :request do
+  around(:example) do |example|
+    OmniAuth.config.test_mode = true
+    example.run
+    OmniAuth.config.mock_auth[:true_layer] = nil
+    OmniAuth.config.test_mode = false
+  end
+
+  let(:token) { SecureRandom.uuid }
+  let(:expires_at) { 1.hour.from_now.round }
+  let(:true_layer_expires_at) { expires_at.to_i }
+  let(:applicant) { create :applicant }
+  let(:identification_tool) { double applicant: applicant }
+
+  before do
+    OmniAuth.config.add_mock(
+      :true_layer,
+      credentials: {
+        token: token,
+        expires_at: true_layer_expires_at
+      }
+    )
+    allow(TrueLayer::IdentifyApplicant)
+      .to receive(:new)
+      .with(token)
+      .and_return(identification_tool)
+  end
+
+  describe 'GET /applicants/auth/true_layer/callback' do
+    subject { get applicant_true_layer_omniauth_callback_path }
+
+    it 'should redirect to next page' do
+      expect(subject).to redirect_to(citizens_accounts_path)
+    end
+
+    it 'should persist token in applicant' do
+      subject
+      expect(applicant.reload.true_layer_token).to eq(token)
+    end
+
+    it 'should persist expires at in applicant' do
+      subject
+      expect(applicant.reload.true_layer_token_expires_at).to eq(expires_at)
+    end
+
+    context 'with a string time' do
+      let(:true_layer_expires_at) { expires_at.to_json }
+      it 'should persist expires at in applicant' do
+        subject
+        expect(applicant.reload.true_layer_token_expires_at).to eq(expires_at)
+      end
+    end
+
+    context 'with nil time' do
+      let(:true_layer_expires_at) { nil }
+      it 'should persist expires at in applicant' do
+        subject
+        expect(applicant.reload.true_layer_token_expires_at).to be_nil
+      end
+    end
+
+    context 'on lookup failure' do
+      let(:identification_tool) { double applicant: nil, error: 'foo' }
```

Comment:
[Ditto](https://github.com/ministryofjustice/laa-apply-for-legal-aid/pull/46/files#r231114895)